dspace-api/src/main/java/org/dspace/content/service/ItemService.java

```diff
@@ -55,6 +55,20 @@
      */
     public Item create(Context context, WorkspaceItem workspaceItem) throws SQLException, AuthorizeException;
 
+    /**
+     * Create a new item, with a provided ID. This method is not public,
+     * since items need to be created as workspace items. Authorisation is the
+     * responsibility of the caller.
+     *
+     * @param context DSpace context object
+     * @param workspaceItem in progress workspace item
+     * @param uuid the pre-determined UUID to assign to the new item
+     * @return the newly created item
+     * @throws SQLException if database error
+     * @throws AuthorizeException if authorization error
+     */
+    public Item create(Context context, WorkspaceItem workspaceItem, UUID uuid) throws SQLException, AuthorizeException;
```

Comment:
This method is public despite claiming to not be public in the JavaDocs.  One or the other should be corrected.

Considering this create() method does NOT check authorization properly, it probably shouldn't be public -- otherwise we are potentially creating security issues if this is used inappropriately.

Reply:
So for this I had just copied the above javadocs and edited it slightly so I figured if this was the case both methods would need to be changed so in the commit I just pushed I put the check for auth in both of these methods and then changed the javadoc to reflect that aswell as the fact that they are public.